optimum/neuron/utils/cache_utils.py

```diff
@@ -719,7 +717,10 @@ def get_cached_model_on_the_hub(neuron_hash: NeuronHash) -> Optional[CachedModel
             repo_id, revision = repo_id
         else:
             revision = "main"
-        repo_filenames = HfApi().list_repo_files(repo_id, revision=revision, token=HfFolder.get_token())
+        try:
+            repo_filenames = HfApi().list_repo_files(repo_id, revision=revision, token=HfFolder.get_token())
+        except Exception:
+            continue
```

Comment:
Do you really need this or was it required while you were investigating issues ?

Reply:
It is needed. 
Basically we are trying to list the files in the `repo_id`. But the `repo` might be private to us, which would lead to an error. 
Basically it is enforced when setting the main cache repo that it is not private to the user, but others might be (usually when playing around the cache repo during testing).